lib/utils/manager/download_manager.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:dio/dio.dart';
+import 'package:fluffychat/data/network/media/media_api.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/utils/manager/downloading_worker_queue.dart';
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/utils/task_queue/task.dart';
+import 'package:matrix/matrix.dart';
+
+typedef FutureVoidCallback = Future<void> Function();
+
+class DownloadManager {
+  DownloadManager._();
+
+  static final DownloadManager _instance = DownloadManager._();
+
+  factory DownloadManager() => _instance;
+
+  final workingQueue = getIt.get<DownloadWorkerQueue>();
+
+  final mediaAPI = getIt.get<MediaAPI>();
+
+  final Map<String, CancelToken> eventIdMapCancelToken = {};
+
+  void cancelDownload(String eventId) {
+    final cancelToken = eventIdMapCancelToken[eventId];
+    if (cancelToken != null) {
+      cancelToken.cancel();
+      eventIdMapCancelToken.remove(eventId);
+    }
+  }
+
+  Future<void> download(
+    Uri uriPath, {
+    required String savePath,
+    required Event event,
+    void Function(int, int)? onDownloadProgress,
```

Comment:
need more info for download progress: `Initial -> Processing (percentage) -> Completed or Error`